cmd/ssh.go

```diff
@@ -248,8 +380,34 @@ func (cmd *SSHCmd) startTunnel(ctx context.Context, devPodConfig *config.Config,
 	writer := log.ErrorStreamOnly().Writer(logrus.InfoLevel, false)
 	defer writer.Close()
 
+	if cmd.GPGAgentForwarding ||
+		devPodConfig.ContextOption(config.ContextOptionGPGAgentForwarding) == "true" {
+		// Check if a forwarding is already enabled and running, in that case
+		// we skip the forwarding and keep using the original one
+		command := "gpg -K"
+		if cmd.User != "" && cmd.User != "root" {
+			command = fmt.Sprintf("su -c \"%s\" '%s'", command, cmd.User)
+		}
+
+		// capture the output, if it's empty it means we don't have gpg-forwarding
+		var out bytes.Buffer
+		err := devssh.Run(ctx, containerClient, command, nil, &out, writer)
```

Comment:
maybe encapsulate it somehow, with `remoteGpgClient`? For me it's too low-level abstraction we use for the function which is responsible for checking if gpg forwarding is already set up and if not then to do it.

Reply:
Moved to the gpg package :+1: